### ISSUE TITLE
Pull from SSH Remote training

### ISSUE THREAD

**Issue by wheatdog** [guildai/guildai#57]
I found that I cannot `guild pull` unless I specify flag `-y`. Here is an example.

Following are contents of source files and guild status,
```
$ guild check

guild_version:             0.6.7.dev1
guild_install_location:    /home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild
guild_home:                /home/wheatdog/miniconda3/envs/guild-test/.guild
guild_resource_cache:      /home/wheatdog/miniconda3/envs/guild-test/.guild/cache/resources
installed_plugins:         cpu, disk, exec_script, gpu, keras, memory, perf, python_script, skopt
python_version:            3.7.4 (default, Aug 13 2019, 20:35:49) [GCC 7.3.0]
python_exe:                /home/wheatdog/miniconda3/envs/guild-test/bin/python
platform:                  Linux 5.2.14-arch2-1-ARCH x86_64
tensorboard_version:       1.15.0
cuda_version:              not installed
nvidia_smi_version:        not installed
latest_guild_version:      0.6.6

$ cat echo.py

print('test')

$ cat guild.yml

- model: test
  operations:
    echo:
      main: echo

$ cat ~/.guild/config.yml

remotes:
    gs203:
        type: ssh
        host: gs203
        guild-home: /home/ytliou/conda/envs/guildai/.guild
        conda-env: guildai
        use-prerelease: True
```

And following are the steps to reproduce,
```
$ guild run -r gs203

...
Run 185d8027 stopped with a status of 'completed'

$ guild runs -r gs203

[1:185d8027]  test/test:echo  2019-09-15 15:25:59  completed

$ guild runs pull gs203

Getting remote run info
Traceback (most recent call last):
  File "/home/wheatdog/miniconda3/envs/guild-test/bin/guild", line 10, in <module>
    sys.exit(main())
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/main_bootstrap.py", line 38, in main
    guild.main.main()
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/main.py", line 32, in main
    _main()
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/main.py", line 38, in _main
    main_cmd.main(standalone_mode=False)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/external/click/core.py", line 759, in __call__
    return self.main(*args, **kwargs)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/external/click/core.py", line 714, in main
    rv = self.invoke(ctx)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/external/click/core.py", line 1132, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/external/click/core.py", line 1132, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/external/click/core.py", line 951, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/external/click/core.py", line 552, in invoke
    return callback(*args, **kwargs)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/external/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/click_util.py", line 146, in fn
    return fn0(*(args + (Args(**kw),)))
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/commands/runs_pull.py", line 77, in pull_runs
    runs_impl.pull(args, ctx)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/commands/runs_impl.py", line 847, in pull
    filtered_runs_f)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/commands/runs_impl.py", line 432, in runs_op
    formatted = formatted = format_runs(selected)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/commands/runs_impl.py", line 357, in format_runs
    _apply_op_desc(formatted)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/commands/runs_impl.py", line 362, in _apply_op_desc
    op_desc = _op_desc_base(fmt_run)
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/commands/runs_impl.py", line 368, in _op_desc_base
    op_dir = _run_op_dir(fmt_run["_run"])
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/commands/runs_impl.py", line 374, in _run_op_dir
    run = _batch_proto(run) or run
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/commands/runs_impl.py", line 384, in _batch_proto
    proto_dir = run.guild_path("proto")
  File "/home/wheatdog/miniconda3/envs/guild-test/lib/python3.7/site-packages/guild/remote.py", line 100, in guild_path
    raise TypeError("guild_path not supported by %s" % self)
TypeError: guild_path not supported by <guild.remote.RunProxy '185d8027d2ca4ec78f3c461e52c4e3c4'>

$ guild runs pull gs203 -y

Getting remote run info
Copying 185d8027d2ca4ec78f3c461e52c4e3c4
receiving incremental file list
created directory /home/wheatdog/miniconda3/envs/guild-test/.guild/runs/185d8027d2ca4ec78f3c461e52c4e3c4
./
.guild/
.guild/opref
.guild/output
.guild/output.index
.guild/attrs/
.guild/attrs/cmd
.guild/attrs/deps
.guild/attrs/env
.guild/attrs/exit_status
.guild/attrs/flags
.guild/attrs/host
.guild/attrs/id
.guild/attrs/initialized
.guild/attrs/opdef
.guild/attrs/opref
.guild/attrs/platform
.guild/attrs/random_seed
.guild/attrs/run_params
.guild/attrs/sourcecode_digest
.guild/attrs/started
.guild/attrs/stopped
.guild/attrs/user

sent 462 bytes  received 4,299 bytes  1,904.40 bytes/sec
total size is 2,865  speedup is 0.60

$ guild runs info

id: 185d8027d2ca4ec78f3c461e52c4e3c4
operation: test/test:echo
from: test==0.0.0
status: completed
started: 2019-09-15 15:25:59
stopped: 2019-09-15 15:26:00
marked: no
label:
sourcecode_digest:
run_dir: ~/miniconda3/envs/guild-test/.guild/runs/185d8027d2ca4ec78f3c461e52c4e3c4
command: /home/ytliou/conda/envs/guildai/bin/python -um guild.op_main echo --
exit_status: 0
pid:
opref: pending:? ? ? test:echo
flags:
```

**Comment by wheatdog** [comment]
I ran the same operation locally, and found there are some values unexpectedly different in guild meta data between these runs:
- operation
- from
- sourcecode_digest

I think it would be helpful if guild can keep these meta data same between local and remote, especially sourcecode_digest. Right now, I can't apply `guild diff -s` on remote training runs.

```
$ guild runs

[1:b78e7373]  test:echo       2019-09-15 15:52:38  completed
[2:185d8027]  test/test:echo  2019-09-15 15:25:59  completed

# [1:b78e7373] was run remotely and [2:185d8027] was run locally

$ guild runs info 2

id: 185d8027d2ca4ec78f3c461e52c4e3c4
operation: test/test:echo
from: test==0.0.0
status: completed
started: 2019-09-15 15:25:59
stopped: 2019-09-15 15:26:00
marked: no
label:
sourcecode_digest:
run_dir: ~/miniconda3/envs/guild-test/.guild/runs/185d8027d2ca4ec78f3c461e52c4e3c4
command: /home/ytliou/conda/envs/guildai/bin/python -um guild.op_main echo --
exit_status: 0
pid:
opref: pending:? ? ? test:echo
flags:

$ guild runs info 1

id: b78e7373c6b24e439bd17990536c4e84
operation: test:echo
from: ~/prj/guild_start/guild.yml
status: completed
started: 2019-09-15 15:52:38
stopped: 2019-09-15 15:52:38
marked: no
label:
sourcecode_digest: 514c74294189b8a76bc382a62a849709
run_dir: ~/miniconda3/envs/guild-test/.guild/runs/b78e7373c6b24e439bd17990536c4e84
command: /home/wheatdog/miniconda3/envs/guild-test/bin/python -um guild.op_main echo --
exit_status: 0
pid:
flags:
```

**Comment by gar1t** [comment]
The bug illustrated in the issue here is fixed and pending a release.

I opened #58 and #59 to track issues related to sourcecode digest and remote diffs respectively.

The `from` attribute is technically correct as Guild generates a package when running remotely. The `from` shows where the operation is defined. In the case of the remote, you're seeing the package name.

The same logic applies to the `operation` attribute - when run remotely, the operation reflects the fact that the operation is run on a package rather than from a project (as the case when run locally).

I think that behavior is correct - given the way Guild implements remote operations. I'm not sure if would be a good idea to force those two attributes to be the same, locally vs remote.

The sourcecode digest is definitely an issue - see #58 for my thinking there.

And Guild should support remote diffs - see #59 for that one.

**Comment by gar1t** [comment]
This is fixed in 0.6.7.dev2, which is available as a pre-release.

**Comment by gar1t** [comment]
Didn't mean to close - let's keep this open until it lands in 0.6.7.

**Comment by gar1t** [comment]
Fix is officially available in 0.7.0.
